src/util/__init__.py

```
@@ -143,47 +146,48 @@
-    else:
-        loc_type = QStandardPaths.StandardLocation.DocumentsLocation
-        dir_ = QDir.toNativeSeparators(QStandardPaths.standardLocations(loc_type)[0])
-        try:
-            dir_.encode("ascii")
+def get_personal_dir() -> str:
+    loc_type = QStandardPaths.StandardLocation.DocumentsLocation
+    doc_dir = QDir.toNativeSeparators(QStandardPaths.standardLocations(loc_type)[0])
+    try:
+        doc_dir.encode("ascii")
+        if not os.path.isdir(doc_dir):
+            raise OSError("No documents location. Will use APPDATA instead.")
```

Ruff (0.14.5)

Abstract `raise` to an inner function

(TRY301)